### PR TITLE
Add config for golangci-lint metalinter (#48)

### LINKED ISSUE #48: Setup linter configuration
#### Motivation / Context

To setup a linter configuration.

#### Description

`golangci-lint` is a metalinter that can be configured to run many linters in parallel. Use the [official template file](https://raw.githubusercontent.com/golangci/golangci-lint/master/.golangci.example.yml) to setup a configuration for running golangci-lint for this project.

- [x]  Propose the linters that can be used.
- [x] Setup linter configuration.
- [ ] Resolve linters errors, if any.

#### Testing

-NA-

#### Thread Safety

-NA-

### PR DESCRIPTION
**Pull request by manoranjith**
#### Description

Add configuration for `golangci-lint` using the linters as proposed in the issue #48.

Version used: `1.28.2 built from 4fb9fe5 on 2020-07-09T23:29:18Z`. 
It can be installed using the command given in the `Testing` section. 

##### Category

Improvement

##### Relevant issue

Resolves #48 
##### Steps to run the tests

1. Install the required version `golangci-lint` using the below command:
```
curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.28.2
```
2. Run `golangci-lint -v`, output shows the `.golangci.yml` in project root is used.

#### Checklist 

- [x] Name is added to the NOTICE file, if it is not present already.
- [x] Changes are rebased onto the target branch.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+  errcheck:
+    # report about assignment of errors to blank identifiers, default is false.
+    check-blank: true
+
+  depguard:
+    list-type: blacklist
+    include-go-root: true
+    packages-with-error-message:
+      - github.com/ethereum/go-ethereum: "GPL licensed code, usage is allowed only inside dst-go/blockchain/ethereum"
```
Comment (choeppler) [inline .golangci.yml:68]: Isn't it LGPL?

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+      - perun.network/go-perun/backend/ethereum:  "uses GPL licensed code, usage is allowed only inside dst-go/blockchain/ethereum"
```
Comment (choeppler) [inline .golangci.yml:69]: Isn't it LGPL

Comment (manoranjith) [reply .golangci.yml]: I have updated the text.

File: .golangci.yml (hunk 1 above)
Comment (manoranjith) [reply .golangci.yml]: Updated.

File: .golangci.yml [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+  errcheck:
+    # report about assignment of errors to blank identifiers, default is false.
+    check-blank: true
+
+  depguard:
+    list-type: blacklist
+    include-go-root: true
+    packages-with-error-message:
+      - github.com/ethereum/go-ethereum: "It contains code licensed under LGPL , import is allowed only inside dst-go/blockchain/ethereum"
```
Comment (choeppler) [inline .golangci.yml:68]:
```suggestion
      - github.com/ethereum/go-ethereum: "It contains code licensed under LGPL, import is allowed only inside dst-go/blockchain/ethereum"
```
fix typo

**Pull request merged**